src/app/scripts/cac/map/cac-map-control.js

```diff
@@ -350,6 +361,9 @@ CAC.Map.Control = (function ($, Handlebars, cartodb, L, turf, _) {
             zoomControl.addTo(map);
             initializeOverlays();
             initializeLayerControl.apply(this, null);
+            if (currentLocationMarker) {
+                currentLocationMarker.addTo(map);
+            }
```

Comment:
In narrow screen widths, the map doesn't initialize on the home page, but only on the map view. In that case the location marker may be initialized before the map, so the check here is needed. I'll see if I can reproduce the marker not appearing.